runtime/utils/common/src/aws/smithy/kotlin/runtime/util/Platform.kt

```diff
@@ -9,16 +9,41 @@ package aws.smithy.kotlin.runtime.util
 public expect object Platform {
     /**
      * Get an environment variable or null
+     *
+     * @param key name of environment variable
+     * @return value of environment variable or null if undefined
      */
     fun getenv(key: String): String?
 
+    /**
+     * Get a system property (on supported platforms)
+     *
+     * @param key name of environment variable
+     * @return value of system property or null if undefined or platform does not support properties
+     */
+    fun getProperty(key: String): String?
+
     val isJvm: Boolean
     val isAndroid: Boolean
     val isBrowser: Boolean
     val isNode: Boolean
     val isNative: Boolean
 
+    /**
+     * The delimiter of segments in a path. For example in Linux: /home/user/documents
+     * or Windows: C:\Program Files\Notepad.EXE
+     */
+    val filePathSegment: String
+
     fun osInfo(): OperatingSystem
+
+    /**
+     * Read the contents of a file as a [String] or return null on any error.
+     *
+     * @param path fully qualified path encoded specifically to the target platform's filesystem.
+     * @return contents of file or null if error (file does not exist, etc.)
+     */
+    fun readFile(path: String): String?
```

Comment:
correctness

1. Should this be `suspend`? (probably...note you can get an [SdkByteReadChannel](https://github.com/awslabs/smithy-kotlin/blob/3b95dac08414a4bc6a3d19e15b610852a20a18a7/runtime/io/jvm/src/aws/smithy/kotlin/runtime/io/FileChannels.kt#L22) from a file or path already)
2. Should we really return `null` on error? This seems likely to hide issues that could be difficult to track down.

Reply:
I don't have any use cases other than the functionality covered by this PR for reading files.  As I mention in the notes above based on this I made this to support my use case and nothing else in the spirit of YAGNI.

What value does adding `suspend` do here?  In the JVM implementation we only call `File.readText()` which does not suspend.  I would be surprised if there is suspend equivalents from JS or native but maybe I'm missing something?

I return null on errors because that maps directly to the requirements of the spec. (`Inaccessible File: If a file is not found or cannot be opened in the configured location, the implementation must treat it as an empty file, and must not attempt to fall back to any other location.`)   If you'd like some other behavior I'd like to see specific use cases.  I'm not saying returning `null` is the right answer here, but I don't want to come up w/ something on a whim.  It seems that when this second use case for file loading comes around we'll need to re-address this and update as needed.  Or, if this turns out to be the only place we need to read files then it's probably fine as is.

Reply:
> I don't have any use cases other than the functionality covered by this PR for reading files. As I mention in the notes above based on this I made this to support my use case and nothing else in the spirit of YAGNI.

I think that adding single specific use cases to a general purpose utility is perhaps not the right way to proceed. If semantics are baked in to apply to a single use case then the next time we need to read a file for any reason and we update this code because it wasn't made general purpose we may break that single use case. 

> What value does adding suspend do here? In the JVM implementation we only call File.readText() which does not suspend. I would be surprised if there is suspend equivalents from JS or native but maybe I'm missing something?

Where is this going to be called from? If the answer is anywhere from an operation then we've just broken the concurrency model OR we move the burden of the blocking call into the call site. Coroutines aren't supposed to block and this introduces a blocking IO call. My suggestion was to remove `File.readText()` in favor of a suspending equivalent OR at the very least run it on the appropriate dispatcher for the caller:

```kotlin
// JVM implementation

actual suspend fun readFileOrNull(path: String): ByteArray? = try {
         with(Dispatchers.IO) {
             File.readBytes(path)
         }
    }catch (ex: IOException) {
        null
     }
```

This is better to me for a few reasons:
* `readFileOrNull()` name matches semantics elsewhere `orNull` is used which usually implies errors are swallowed/ignored
* it returns a `ByteArray?` rather than `String?` which makes this more general purpose. Now we can read binary files if needed. Callers that want text only have to call `decodeToString()`
* It is usable from a coroutine and callers don't need to think about the blocking call or misusing it accidentally

Leave it as is for now if you feel it's right but I personally hate adding a public API without thinking through how it will/could be used. YAGNI doesn't really apply here IMHO.

Reply:
> If semantics are baked in to apply to a single use case then the next time we need to read a file for any reason and we update this code because it wasn't made general purpose we may break that single use case.

I think this is the general nature of development.  Also, this code is internal.

> remove File.readText() in favor of a suspending equivalent 

I'm not aware of a JVM suspending equivalent.  I notice something about Java NIO w/ channels but seems overly complex and I can't think of a concrete benefit.  LMK if you know of something here.

Otherwise, your suggestions and code snippets are good and thank you for raising the concurrency concerns regarding blocking calls in coroutines.  I'll update the PR to use your suggestion.

Reply:
> I'm not aware of a JVM suspending equivalent.

FWIW we have [wrappers](https://github.com/awslabs/smithy-kotlin/blob/main/runtime/io/jvm/src/aws/smithy/kotlin/runtime/io/FileChannels.kt) in our runtime to create `SdkByteReadChannels`.

The suggestion I made is fine and more or less equivalent though (file channels are moved to `Dispatchers.IO` internally as well for all the blocking reads).